src/Avalonia.Controls/ApplicationLifetimes/ClassicDesktopStyleApplicationLifetime.cs

```diff
@@ -132,6 +132,26 @@ public int Start(string[] args)
             return _exitCode;
         }
 
+        /// <summary>
+        /// Since the lifetime must be set up/prepared with 'args' before executing Start(), an overload with no parameters seems more suitable for integrating with some lifetime manager providers, such as MS HostApplicationBuilder.
+        /// </summary>
+        /// <returns></returns>
+        public int Start()
+        {
+            SetupCore(Args ?? Array.Empty<string>());
+
+            _cts = new CancellationTokenSource();
+
+            // Note due to a bug in the JIT we wrap this in a method, otherwise MainWindow
+            // gets stuffed into a local var and can not be GCed until after the program stops.
+            // this method never exits until program end.
+            ShowMainWindow();
+
+            Dispatcher.UIThread.MainLoop(_cts.Token);
+            Environment.ExitCode = _exitCode;
+            return _exitCode;
```

Comment:
Maybe extract this part into StartCore or something like that? Otherwise, it repeats important part of the lifetime code, even with comments.

Reply:
Good advice, I have updated this pr.

Reply:
OK, I have updated this pr.